### PR TITLE
Feature/select2 ajax option filters

### LINKED ISSUE #4444: select2 minimumInputLength (Performance Improvement)
I need a way to set the minimumInputLength per field ( can be a hook )

`                minimumInputLength : 1
`
Default value is 1, and with a large number of posts, it takes a lot of time.

### PR DESCRIPTION
**Pull request by pglewis**
Fixes #4444 and #4445

### CODE REVIEW

**Comment by pglewis** [comment]
This is just a minimal patch to support this for 2.7 release.  @jamesgol has a more flexible approach that will support arbitrary select2 options to be customized via filter, I'm just too paranoid to jostle how we work with the JS options this close to release.  

For now, this will do the trick: 

```add_filter( 'pods_field_dfv_data', 'set_select2_options', 10, 2 );
function set_ajax_delay( $data, $args ) {
	$data['fieldConfig']['ajax_data']['delay'] = 1000;
	$data['fieldConfig']['ajax_data']['minimum_input_length'] = 3;

	return $data;
}```

You can just opt-in when we re-organize how we do this and the above should continue to work either way.

File: classes/fields/pick.php [hunk 1]
```diff
@@ -872,7 +872,7 @@ public function build_dfv_field_options( $options, $args ) {
 
 		$options[ $args->type . '_limit' ] = $limit;
 
-		$options[ 'ajax_data' ] = $this->build_dfv_autocomplate_ajax_data( $options, $args, $ajax );
+		$options[ 'ajax_data' ] = $this->build_dfv_autocomplete_ajax_data( $options, $args, $ajax );
```
Comment (sc0ttkclark) [inline classes/fields/pick.php]: That's a weird diff

File: classes/fields/pick.php [hunk 2]
```diff
@@ -895,7 +895,7 @@ public function build_dfv_field_options( $options, $args ) {
 	 *
 	 * @return array
 	 */
-	public function build_dfv_autocomplate_ajax_data( $options, $args, $ajax ) {
+	public function build_dfv_autocomplete_ajax_data( $options, $args, $ajax ) {
```
Comment (sc0ttkclark) [inline classes/fields/pick.php]: That's a weird diff

File: classes/fields/pick.php (hunk 1 above)
Comment (pglewis) [reply classes/fields/pick.php]: I may have made some mention about reading glasses....

**Pull request merged**